src/util.lisp

```diff
@@ -85,14 +85,16 @@
   (ensure-directories-exist directory)
   (if (probe-file (fad:merge-pathnames-as-directory directory ".git/"))
       (let (stash
+            ;; https://stackoverflow.com/a/15284176
+            (upstream-ref (first (run-git-command "rev-parse --symbolic-full-name @{u}" directory)))
             (pre-revision (first (run-git-command "rev-parse HEAD" directory))))
         (format t "~%---> Upgrading ~A" name)
         (run-git-command "fetch origin" directory)
 
         (setq stash (first (run-git-command "status --untracked-files=all --porcelain" directory)))
         (if stash (run-git-command "stash" directory))
 
-        (run-git-command "rebase origin/master" directory)
+        (run-git-command (concatenate 'string "rebase " upstream-ref) directory)
```

Comment:
This means that `update-repository` now requires that the repo has an upstream set.  

I made this change because it's possible that a repository doesn't have a master branch (`yason` doesn't, for example).

At https://github.com/CodyReichert/qi/pull/41/files#diff-4c07d3530c3fc7ab6a28f422602b753bR118 I updated the code to set an upstream when it initializes a repository, but if that code previously ran to create a repository for the manifest, then it won't have an upstream set, and Qi will error out when it tries to update the manifest.

This can be fixed by manually running `git branch --set-upstream-to=origin/master master` in the manifest repository.